packages/devextreme/scss/widgets/base/treeView/_index.scss

```diff
@@ -1,71 +1,175 @@
 .dx-treeview-custom-collapse-icon {
-  position: absolute;
-  left: 0;
-  top: 0;
   cursor: pointer;
 }
 
-@mixin dx-treeview-make-rtl($container-with-border-offset, $node-offset, $item-with-checkbox-offset, $checkbox-offset, $select-all-text-offset, $select-all-item-left-padding, $border-padding) {
-  &.dx-treeview-border-visible {
-    .dx-treeview-select-all-item {
-      padding-left: 0;
-      padding-right: $select-all-item-left-padding;
-    }
+@mixin default-expander-icon($expand-icon-name, $collapse-icon-name, $font-size, $size, $color) {
+  .dx-treeview-toggle-item-visibility {
+    @include dx-icon($collapse-icon-name);
 
-    .dx-scrollable-content {
-      > .dx-treeview-node-container {
-        padding-left: $border-padding;
-        padding-right: $container-with-border-offset;
-      }
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    font-size: $font-size;
+    width: $size;
+    height: $size;
+    color: $color;
+
+    &.dx-treeview-toggle-item-visibility-opened {
+      @include dx-icon($expand-icon-name);
+
+      font-size: $font-size;
     }
   }
 
-  .dx-treeview-node {
-    padding-right: $node-offset;
+  .dx-treeview-expander-icon-stub {
+    display: flex;
+    flex-shrink: 0;
+    width: $size;
+    height: $size;
   }
+}
 
+@mixin treeview-item($gap, $padding, $icon-size, $hover-bg, $item-hover-color, $item-selected-color, $node-padding, $margin: 0, $border-radius: 0) {
   .dx-treeview-item {
+    gap: $gap;
+    padding: $padding;
+    margin: $margin;
+    
     .dx-treeview-item-content {
       .dx-icon {
-        margin-left: 5px;
+        @include dx-icon-sizing($icon-size);
+  
+        vertical-align: bottom;
       }
     }
+  
+    &.dx-state-hover {
+      background-color: $hover-bg;
+      border-radius: $border-radius;
+      color: $item-hover-color;
+    }
   }
 
-  .dx-treeview-item-with-checkbox {
-    .dx-treeview-item {
-      padding-right: $item-with-checkbox-offset;
+  .dx-state-selected {
+    > .dx-treeview-item {
+      color: $item-selected-color;
     }
+  }
 
-    .dx-checkbox {
-      right: $checkbox-offset;
-    }
+  .dx-treeview-node {
+    padding-inline-start: math.div($node-padding, 2);
   }
+}
 
+@mixin treeview-aux-items(
+  $searchbox-margin-bottom,
+  $search-editor-height,
+  $border-color,
+  $select-all-item-padding,
+  $checkbox-icon-size,
+  $item-element-gap,
+  $select-all-item-border-visible-padding,
+  $treeview-container-padding,
+  $load-indicator-size,
+) {
+  .dx-treeview-search {
+    margin-bottom: $searchbox-margin-bottom;
+  }
+  
+  .dx-treeview-with-search > .dx-scrollable {
+    height: calc(100% - #{round($search-editor-height + $searchbox-margin-bottom)});
+  }
+  
   .dx-treeview-select-all-item {
-    padding-left: 0;
-    padding-right: $checkbox-offset;
-
+    border-block-end: 1px solid $border-color;
+    padding: $select-all-item-padding;
+  
+    .dx-checkbox-container {
+      .dx-checkbox-icon {
```

Comment:
It seems to me that for the select all checkbox we should not configure sizes different from the default sizes of the checkbox component